lib/create-contentful-api.ts

```diff
@@ -520,8 +520,8 @@ export default function createContentfulApi<OptionType>(
     query: EntryQueries = {}
   ): Promise<EntryWithLinkResolutionAndWithUnresolvableLinks<Fields>> {
     if (query.locale === '*') {
-      console.warn(
-        `If you want to fetch entries in all existing locales, we recommend you to use client.withAllLocales instead of the locale='*' parameter.`
+      console.error(
+        `To fetch an entry in all existing locales use client.withAllLocales instead of the locale='*' parameter.`
       )
     }
```

Comment:
We should throw a ValidationError here (and at the other places):
```js
if (query.locale === '*') {
    throw new ValidationError('locale', 'To fetch an entry in all existing locales use client.withAllLocales instead of the `locale='*'` parameter.')
}
```